docs: Add new DaemonSet FAQ for new DaemonSets deployed after Nodes exist (#4161)
aws/karpenter: website/content/en/docs/faq.md

As an example, suppose you scale up a deployment with a preferred zonal topology spread and none of the newly created pods can run on your existing cluster.  Karpenter will then launch multiple nodes to satisfy that preference.  If a) one of the nodes becomes ready slightly faster than other nodes and b) has enough capacity for multiple pods, `kube-scheduler` will schedule as many pods as possible to the single ready node so they won't remain unschedulable. It doesn't consider the in-flight capacity that will be ready in a few seconds.  If all of the pods fit on the single node, the remaining nodes that Karpenter has launched aren't needed when they become ready and consolidation will delete them.

### When deploying an additional DaemonSet to my cluster, why does Karpenter not scale-up my nodes to support the extra DaemonSet?

> Karpenter will not scale-up more capacity for a DaemonSet since the only pods that would schedule to that new Node that Karpenter would scale _would_ be the DaemonSet alone; therefore, Karpenter only considers DaemonSets when doing overhead calculations for scale-ups to workload pods.

Reviewer: The extra pieces made it a little bit tougher to understand. Consider this:
```suggestion
Karpenter will not scale-up more capacity for an additional DaemonSet since the only pods that would schedule to the new Node _would_ be the DaemonSet alone. Therefore, Karpenter only considers DaemonSets when doing overhead calculations for scale-ups to workload pods.
```

> To avoid new DaemonSets not being scheduled to existing Nodes, you should [set a high priority on your DaemonSet pods with a `preemptionPolicy: PreemptLowerPriority`](https://kubernetes.io/docs/concepts/scheduling-eviction/pod-priority-preemption/#example-priorityclass) so that DaemonSet pods will be guaranteed to schedule on all existing and new Nodes. For existing Nodes, this will cause some pods with lower priority to get [preempted](https://kubernetes.io/docs/concepts/scheduling-eviction/pod-priority-preemption/#preemption), replaced by the DaemonSet and re-scheduled onto new capacity that Karpenter will launch in response to the new pending pods.

Reviewer: consider:
```suggestion
To avoid new DaemonSets failing to schedule to existing Nodes, you should [set a high priority on your DaemonSet pods with a `preemptionPolicy: PreemptLowerPriority`](https://kubernetes.io/docs/concepts/scheduling-eviction/pod-priority-preemption/#example-priorityclass) so that DaemonSet pods will be guaranteed to schedule on all existing and new Nodes. For existing Nodes, this will cause some pods with lower priority to get [preempted](https://kubernetes.io/docs/concepts/scheduling-eviction/pod-priority-preemption/#preemption), replaced by the DaemonSet and re-scheduled onto new capacity that Karpenter will launch in response to the new pending pods.
```
